File: mysql_examples/recipes/transfer/check_enum_value.php
<?php
# check_enum_value.php - wrapper to demonstrate check_enum_value()
# utility function that determines whether a value is a member of
# a given ENUM column.

# Usage: check_enum_value.php db_name tbl_name col_name test_value

require_once "Cookbook.php";
require_once "Cookbook_Utils.php";

if ($argc != 5)
  die ("Usage: check_enum_value.php db_name tbl_name col_name test_value\n");

$db_name = $argv[1];
$tbl_name = $argv[2];
$col_name = $argv[3];
$val = $argv[4];

$conn =& Cookbook::connect ();
if (PEAR::isError ($conn))
  die ("Cannot connect to server: " . $conn->getMessage () . "\n");

$valid = check_enum_value ($conn, $db_name, $tbl_name, $col_name, $val);

print ("$val "
        . ($valid ? "is" : "is not")
        . " a member of $tbl_name.$col_name\n");

$conn->disconnect ();

?>

File: mysql_examples/recipes/transfer/check_enum_value.py
#!/usr/bin/python
# check_enum_value.py - wrapper to demonstrate check_enum_value()
# utility function that determines whether a value is a member of
# a given ENUM column.

# Usage: check_enum_value.py db_name tbl_name col_name test_value

import sys
import MySQLdb
import Cookbook
from Cookbook_Utils import *

if len (sys.argv) != 5:
  print "Usage: check_enum_value.py db_name tbl_name col_name test_val"
  sys.exit (1)
db_name = sys.argv[1]
tbl_name = sys.argv[2]
col_name = sys.argv[3]
val = sys.argv[4]

try:
  conn = Cookbook.connect ()
except MySQLdb.Error, e:
  print "Message:", e.args[1]
  print "Code:", e.args[0]
  sys.exit (1)

valid = check_enum_value (conn, db_name, tbl_name, col_name, val)

print val,
if valid:
  print "is",
else:
  print "is not",
print "a member of " + tbl_name + "." + col_name

conn.close ()
